demisto_sdk/commands/common/hook_validations/structure.py

```diff
@@ -380,3 +379,27 @@ def check_for_spaces_in_file_name(self):
                 return False
 
         return True
+
+
+def checked_type_by_reg(file_path, compared_regexes=None, return_regex=False):
```

Comment:
why put this function in structure? I get that it is the only location it is currently used but is still seems like a tool to me

Reply:
I guess that the main purpose of that is so this function would be a tool only for structure.py and people wont use it for other purposes. right? @bakatzir

Reply:
It does look like a tool func, but we want dev to shy away from it and use `find_type` only.

Reply:
So the idea is to minimize the use of regex, that is a good point. Let's keep it here for now and worse case scenario we'll move it to tools.py if it needs to be used elsewhere